utils/nce_neighbors.py

```diff
@@ -0,0 +1,46 @@
+import random
+import numpy as np
+import heapq
+import operator
+
+import torch
+from torchtext import data
+
+# get the nearest negative samples to the positive sample by computing the feature difference
+def get_nearest_neg_id(pos_feature, neg_dict, distance="cosine", k=1):
+    dis_list = []
+    pos_feature = pos_feature.data.cpu().numpy()
+    pos_feature_norm = pos_feature / np.sqrt(sum(pos_feature ** 2))
+    neg_list = []
+    for key in neg_dict:
+        if distance == "l2":
+            dis = np.sqrt(np.sum((np.array(pos_feature) - neg_dict[key]["feature"]) ** 2))
+        elif distance == "cosine":
+            neg_feature = np.array(neg_dict[key]["feature"])
+            feat_norm = neg_feature / np.sqrt(sum(neg_feature ** 2))
```

Comment:
Use `numpy.linalg.norm`? https://stackoverflow.com/a/1401828/2514889

Reply:
I seldom use `numpy.linalg.norm` or `scipy.linalg.norm` because there is a type issue: https://stackoverflow.com/questions/31685156/normalizing-vector-produces-nan-in-numpy
If I write it myself, I do not worry about the possible mistake, right?